src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.ImportExport.cs

```diff
@@ -38,10 +38,13 @@ public override void ImportParameters(RSAParameters parameters)
             unsafe
             {
                 if (parameters.Exponent == null || parameters.Modulus == null)
-                    throw new ArgumentException(SR.Cryptography_InvalidRsaParameters);
+                    throw new CryptographicException(SR.Cryptography_InvalidRsaParameters);
 
                 bool includePrivate = parameters.P != null && parameters.Q != null;
 
+                if (includePrivate && parameters.DP == null)
```

Comment:
The CSP implementation enforces that they're all present; OpenSSL would function just fine with only D, but since this is a data exchange type/contract it went with the "specify everything" approach; and I feel it's better to have them all match here with the most stringent requirements; otherwise you get weird cross-plat behaviors with different minimum requirements.